### PR TITLE
made sudo call from console with confirmation for previously generated chains

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Issue
https://app.clickup.com/t/2jvu8rt

## Description
- based on https://github.com/ComposableFi/composable/pull/1705
- made sudo call from console with confirmation for previously generated chains
- pd.js does not allow to run sudo from encoded call so it nightmare to do xcmp (not for sudo too)

### CODE REVIEW

File: code/utils/xcmp/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
+use clap::{clap_derive::ArgEnum, Parser, Subcommand};
+
+#[derive(Parser, Debug)]
+#[clap()]
```
Comment (benluelo) [inline code/utils/xcmp/src/config.rs:4]: Is the empty attribute required?

File: code/utils/xcmp/src/main.rs [hunk 2]
```diff
@@ -111,18 +46,90 @@ pub async fn main() {
...
+		Command::Sudo(command) => match command.command {
+			SudoCommand::Execute(execute) =>
+				execute_sudo(execute.ask, execute.call, execute.network, execute.suri, execute.rpc)
+					.await,
+			_ => todo!("implement"),
+		},
+	}
+}
+
+macro_rules! decode_call {
```
Comment (benluelo) [inline code/utils/xcmp/src/main.rs:58]: Could these macros just be functions? `dali` is the only network ever passed to them, and `decode_call!` seems to be hard-coded to dali_runtime anyways?

Comment (dzmitry-lahoda) [reply code/utils/xcmp/src/main.rs:58]: mistake. nees to set picasso.

File: code/utils/xcmp/src/config.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply code/utils/xcmp/src/config.rs:4]: you mean if need clap or if need () in clap?

**Review by alexkdefi — APPROVED** [review]
👍

File: code/utils/xcmp/src/config.rs (hunk 1 above)
Comment (benluelo) [reply code/utils/xcmp/src/config.rs]: If the entire attribute is required - afaik it doesn't do anything

File: code/utils/composable-subxt/subxt.nix [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{ pkgs }:
+pkgs.rustPlatform.buildRustPackage rec {
+  pname = "subxt-cli";
+  # NOTE: keep in sync with Cargo.toml as generator version must be same as dependency to compile generated code
+  version = "0.23.0";
```
Comment (benluelo) [inline code/utils/composable-subxt/subxt.nix:5]: Is there a way to read this from the Cargo.toml file?

Comment (dzmitry-lahoda) [reply code/utils/composable-subxt/subxt.nix:5]: no.

Comment (dzmitry-lahoda) [reply code/utils/composable-subxt/subxt.nix:5]: From local Cargo.toml yes. from remote - now. so I am talking about local - it exists. But not for remote. 
I will not do local in PR as nobody does that, just mention. If somebody will want to try nix.

Comment (dzmitry-lahoda) [reply code/utils/composable-subxt/subxt.nix:5]: closing for now.

File: code/utils/xcmp/src/config.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply code/utils/xcmp/src/config.rs]: removed

**Review by benluelo — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@haroldsphinx i tried 3 times already. i am main rebased this morning

image

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

**Review by andor0 — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

**Review by Martin-Kiss — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

**Review by Dom-Roth — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

**Pull request merged**